src/handle-data.ts

```diff
@@ -58,6 +59,34 @@ export function getTotalInfo(layout: TableLayout, colIdx: number, pageColIdx: nu
   return '';
 }
 
+/**
+ * Gets the totals alignment for head, totals and body
+ * bodyTextAlign is later used to determine independent alignment for each body cell
+ */
+export function getAlignInfo(
+  textAlign: TextAlign,
+  qDimensionType: EngineAPI.DimensionType | undefined,
+  isDim: boolean
+) {
+  const autoHeadTextAlign = qDimensionType === 'N' || qDimensionType === undefined ? 'right' : 'left';
+  const autoTotalsTextAlign = isDim ? 'left' : 'right';
+  const headTextAlign: Align = !textAlign || textAlign.auto ? autoHeadTextAlign : textAlign.align;
+  const totalsTextAlign: Align = !textAlign || textAlign.auto ? autoTotalsTextAlign : textAlign.align;
+  const bodyTextAlign: Align | 'auto' = !textAlign || textAlign.auto ? 'auto' : textAlign.align;
+
+  return { headTextAlign, totalsTextAlign, bodyTextAlign };
+}
+
+/**
+ * Gets the correct text alignment for body cells, based on the text alignment info from the column and cell content
+ */
+export const getBodyCellAlign = (cell: EngineAPI.INxCell, textAlign: Align | 'auto') => {
+  if (textAlign === 'auto') {
+    return ((cell.qNum || cell.qNum === 0) && !Number.isNaN(+cell.qNum) ? 'right' : 'left') as Align;
```

Comment:
```suggestion
    return ((cell.qNum || cell.qNum === 0) && Number.isInteger(cell.qNum) ? 'right' : 'left') as Align;
```
is it always integer?

Reply:
nope can be any number